src/lib/holds.js

```diff
@@ -0,0 +1,60 @@
+'use strict'
+
+const InsufficientFundsError = require('../errors/insufficient-funds-error')
+
+function adjustBalance (account, amount, transaction) {
+  const updateSQL = 'UPDATE "accounts" SET "balance" = "balance" + ? WHERE "name" = ?'
+  /* eslint-disable handle-callback-err */
+  return transaction.raw(updateSQL, [amount, account]).catch((error) => {
+    throw new InsufficientFundsError('Sender has insufficient funds.', account)
+  })
+  /* eslint-enable */
+}
+
+function insertEntry (account, transferID, transaction) {
+  return transaction.select('id')
+    .from('accounts').where('name', account)
+    .then((rows) => {
+      const entry = {
```

Comment:
Let's leave that as a follow-up task because it's not easy to get it working in Oracle